### PR TITLE
AG-227: Explicitly close connection after reconnect for commit

### PR DESCRIPTION
**Pull request by graben**
Hi @barreiro, I think we should add this additional commit to avoid a new introduced leak. Maybe add the same pattern to other methods too. IMHO, none of them is used outside recover START/ END cycle. (Official Narayana modules, of course!)

### CODE REVIEW

**Comment by barreiro** [comment]
@graben right. I was thinking the same when I merged the PR, but was not able to spend any time with that.

File: agroal-narayana/src/main/java/io/agroal/narayana/RecoveryXAResource.java [hunk 1]
```diff
@@ -53,6 +55,11 @@ public Xid[] recover(int flag) throws XAException {
     @Override
     public void commit(Xid xid, boolean onePhase) throws XAException {
         getConnectedResource().commit( xid, onePhase );
+        // Commit can be called after a recover(TMENDRSCAN) has been invoked.
+        // Close after reconnect to avoid a connection leak.
+        if ( reconnected ) {
```
Comment (barreiro) [inline agroal-narayana/src/main/java/io/agroal/narayana/RecoveryXAResource.java:60]: need the same for `rollback()`, no !?

Comment (graben) [reply agroal-narayana/src/main/java/io/agroal/narayana/RecoveryXAResource.java:60]: I did not find a reference on the first check, but double check is on my to-do list. Probably also add another test case which does the real whole roundtrip, but with more Narayana insights, of course.

**Comment by graben** [comment]
@barreiro : I add another test class with an enhanced recovery test case maybe merge into RecoveryTests but it's quite unreadable with so many embedded classes.

**Comment by graben** [comment]
@barreiro : So, I think this could be a leak free implementation ;)

**Review by barreiro — APPROVED** [review]

**Comment by barreiro** [comment]
Great work !! Thanks !!

**Pull request merged**